internal/providers/service.go

```diff
@@ -157,6 +165,123 @@ func (p *providerService) CreateGitHubOAuthProvider(ctx context.Context, provide
 	return &provider, nil
 }
 
+// CreateGitHubAppProvider creates a GitHub App provider with an installation ID
+func (p *providerService) CreateGitHubAppProvider(
+	ctx context.Context,
+	providerClass db.ProviderClass,
+	token oauth2.Token,
+	stateData db.GetProjectIDBySessionStateRow,
+	installationID int64,
+) (*db.Provider, error) {
+	installationOwner, err := p.getInstallationOwner(ctx, installationID)
+	if err != nil {
+		return nil, fmt.Errorf("error getting installation: %w", err)
+	}
+
+	tx, err := p.store.BeginTransaction()
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "error starting transaction: %v", err)
+	}
+	defer p.store.Rollback(tx)
```

Comment:
@dmjb added a `db.WithTransaction` Let's use that instead.

https://github.com/stacklok/minder/blob/main/internal/db/store.go#L118